src/main/java/com/github/princesslana/eriscasper/repository/event/GuildsFromEvents.java

```diff
@@ -0,0 +1,93 @@
+package com.github.princesslana.eriscasper.repository.event;
+
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.event.*;
+import com.github.princesslana.eriscasper.data.resource.Channel;
+import com.github.princesslana.eriscasper.data.resource.Guild;
+import com.github.princesslana.eriscasper.data.resource.ImmutableChannel;
+import com.github.princesslana.eriscasper.repository.GuildRepository;
+import io.reactivex.Maybe;
+import io.reactivex.Observable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Stack;
+
+public class GuildsFromEvents implements GuildRepository {
+
+  private final Map<Snowflake, Guild> guildCache;
```

Comment:
How does this work when there are multiple threads?

My preferred option is to do as `UsersFromEvents`, we maintain the state using the `scan` methods in Observable, just because it fits with the function/reactive/immutable style.

Alternatively, this should at least be changed to a `ConcurrentMap`.